### DISCUSSION TITLE
[Feature] Additional shapes

### DISCUSSION THREAD

**Discussion by altavir** [FXyz/FXyz#163] [Announcements]
We are currently working on a visualization package for simulations in particle physics and considering JavaFX as a possible backend alongside Three.js. We can't rely on exports and imports since we need all elements to be interactive. We need to support at least basic elements from [GDML geometry definition](http://lcgapp.cern.ch/project/simu/framework/GDML/doc/GDMLmanual.pdf), but I could not find some of them in `FXyz`. Specifically, for the first task we need following:

- [ ] Logical operations helper (union, subtraction).

- [ ] Tube

- [ ] Extruded 2d shape helper

- [ ] Polyhedron. Not sure about API, GDML uses layer-based approach.

**Comment by Birdasaur** [comment]
hello altavir.  I believe we currently have or are very close to having the functionality to support some of these already.  Can you point us to some example images of each of these so that we fully understand what you need?

**Comment by altavir** [comment]
It is good to hear. In the end, we need to support the whole GDML specification (the link above). For now, we have a specific accelerator experiment with a fixed deadline (and it has only the shapes I mentioned). I will have to ask guys to create example images for different primitives shapes.

**Comment by altavir** [comment]
[G4Geo.pdf](https://github.com/FXyz/FXyz/files/2936451/G4Geo.pdf)
Here is the part of the manual with pictures. We do not need all of them immediately.

**Comment by Birdasaur** [comment]
Have you had a chance to experiment with using the existing components for your needs yet?
jperedadnr just pushed a new version, 0.5.0, which has a bunch of nice upgrades, some of which would help your use case I believe.

**Comment by altavir** [comment]
We had to put it on hold and focus on JS implementation. We still plan to have JavaFX implementation somewhere in future, but first we need to make working prototype.
